lib/test/Test/Pos/Cbor/CborSpec.hs

```diff
@@ -71,124 +66,22 @@ import           Test.Pos.Util.QuickCheck (SmallGenerator)
 type VoteId' = Tagged U.UpdateVote U.VoteId
 type UpId' = Tagged (U.UpdateProposal, [U.UpdateVote])U.UpId
 
-
-data MyScript = MyScript
-    { version :: ScriptVersion -- ^ Version
-    , script  :: ByteString    -- ^ Serialized script
-    } deriving (Eq, Show, Generic, Typeable)
-
-instance Arbitrary MyScript where
-    arbitrary = MyScript <$> arbitrary <*> arbitrary
-
-deriveSimpleBi ''MyScript [
-    Cons 'MyScript [
-        Field [| version :: ScriptVersion |],
-        Field [| script  :: ByteString   |]
-    ]]
-
-----------------------------------------
-
-data X1 = X1 { x1A :: Int }
-    deriving (Eq, Ord, Show, Generic)
-
-data X2 = X2 { x2A :: Int, x2B :: String }
-    deriving (Eq, Ord, Show, Generic)
-
-instance Arbitrary X1 where
-    arbitrary = genericArbitrary
-    shrink = genericShrink
-
-instance Arbitrary X2 where
-    arbitrary = genericArbitrary
-    shrink = genericShrink
-
-instance Bi (Attributes X1) where
-    encode = encodeAttributes [(0, serialize . x1A)]
-    decode = decodeAttributes (X1 0) $ \n v acc -> case n of
-        0 -> pure $ Just $ acc { x1A = unsafeDeserialize v }
-        _ -> pure $ Nothing
-
-instance Bi (Attributes X2) where
-    encode = encodeAttributes [(0, serialize . x2A), (1, serialize . x2B)]
-    decode = decodeAttributes (X2 0 []) $ \n v acc -> case n of
-        0 -> return $ Just $ acc { x2A = unsafeDeserialize v }
-        1 -> return $ Just $ acc { x2B = unsafeDeserialize v }
-        _ -> return $ Nothing
-
 ----------------------------------------
 
-soundSerializationAttributesOfAsProperty
-    :: forall a b aa ab. (aa ~ Attributes a, ab ~ Attributes b,
-                          Bi aa, Bi ab, Eq aa, Arbitrary a, Show aa)
-    => Property
-soundSerializationAttributesOfAsProperty = forAll arbitraryAttrs $ \input ->
-    let serialized      = serialize input
-        (middle  :: ab) = unsafeDeserialize serialized
-        (encoded :: aa) = unsafeDeserialize $ serialize middle
-    in encoded === input
-  where
-    arbitraryAttrs :: Gen aa
-    arbitraryAttrs = Attributes <$> arbitrary <*> arbitrary
-
-
 spec :: Spec
 spec = withDefConfiguration $ \_ -> do
     describe "Cbor.Bi instances" $ do
         modifyMaxSuccess (const 1000) $ do
-            describe "Test instances" $ do
-                binaryTest @MyScript
-                prop "X2" (soundSerializationAttributesOfAsProperty @X2 @X1)
             describe "Lib/core instances" $ do
-                binaryTest @(Attributes X1)
-                binaryTest @(Attributes X2)
                 brokenDisabled $ binaryTest @UserSecret
                 modifyMaxSuccess (min 50) $ do
                     binaryTest @WalletUserSecret
                     binaryTest @EncryptedSecretKey
```

Comment:
I'm curious why some of the `binaryTest` were moved but not others. I suppose the moved tests were for types defined in `core` and these ones are defined in `lib`?

Reply:
The `WalletUserSecret` is not defined in `core`, its in `crypto`. Doing it in many small PRs rather than one big one. `crypto` is currently a WIP.